java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -230,24 +229,16 @@
         }
         else if (expression instanceof PsiMethodReferenceExpression) {
           List<PsiType> types = new ArrayList<>();
-          PsiType[] typeArguments = ((PsiMethodReferenceExpression)expression).getTypeParameters();
-          for (JavaResolveResult result : ((PsiMethodReferenceExpression)expression).multiResolve(true)) {
+          final PsiMethodReferenceExpression referenceExpression = (PsiMethodReferenceExpression)expression;
+          for (JavaResolveResult result : (referenceExpression).multiResolve(true)) {
             final PsiElement element = result.getElement();
+            if (element == null) continue;
+
             if (element instanceof PsiMethod) {
               PsiMethod method = (PsiMethod)element;
-              int offset = hasOffset((PsiMethodReferenceExpression)expression, method) ? 1 : 0;
+              int offset = hasOffset(referenceExpression, method) ? 1 : 0;
               final PsiParameter[] targetMethodParameters = method.getParameterList().getParameters();
               if (targetMethodParameters.length + offset == parameters.length) {
-                PsiSubstitutor partialSubstitutor = PsiSubstitutor.EMPTY;
```

Comment:
I removed this part because I noticed that `result.getSubstitotor()` already contains the proper substitution map.